core/src/main/java/io/fabric8/maven/core/access/ClusterAccess.java

```diff
@@ -72,6 +73,18 @@ public String getNamespace() {
         return namespace;
     }
 
+    /**
+     * Returns true if this cluster is a traditional OpenShift cluster with the <code>/oapi</code> REST API
+     * or supports the new <code>/apis/image.openshift.io</code> API Group
+     */
```

Comment:
We should move away from the idea that there's a boolean for whether a cluster is openshift or not really. At some point OpenShift will be a set of microservices that can run on any kubernetes cluster.

Right now we can easily add the `build.openshift.io` API Group to support `BuildConfig` and `Build` resources on kubernetes 1.7 so that we can support the jenkins sync plugin and the fabric8 / openshift.io console on vanilla kubernetes for Jenkins pipelines. There's no S2I (as thats supported via `image.openshift.io` API Group which isn't gonna run on kubernetes any time soon I suspect) but at least we can work with BuildConfig/Build resources from a Jenkins pipeline perspective in a consistent way between kubernetes and today's openshift clusters.